### ISSUE TITLE
[BUG] [EKS] Unable to re-import a deleted cluster

### ISSUE THREAD

**Issue by valaparthvi** [rancher/rancher#47863]
**Rancher Server Setup**
- Rancher version: 2.10.0-alpha6
- Installation option (Docker install/Helm Chart):
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): K3s
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: 1.30
- Cluster Type (Local/Downstream):
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Hosted provider EKS

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions:
Admin

**Describe the bug**

Re-importing a deleted cluster returns the following error - 
```
Failed to communicate with cluster: error creating service account: serviceaccounts "kontainer-engine" is forbidden: unable to create new content in namespace cattle-system because it is being terminated
```

**To Reproduce**

1. Import a cluster and delete it
2. Wait for 5-7 min
3. Re-import the cluster

**Result**
Unable to re-import the cluster. Seeing this error - 
```
Failed to communicate with cluster: error creating service account: serviceaccounts "kontainer-engine" is forbidden: unable to create new content in namespace cattle-system because it is being terminated
```
**Expected Result**

The cluster should be re-imported without any error.
**Screenshots**

![image](https://github.com/user-attachments/assets/628a7017-cd1e-4a45-8995-fbb2c8d3b2f3)

**Additional context**

**Comment by cpinjani** [comment]
It seems `cattle-system` namespace remains in Terminating state on downstream cluster after it is deleted in Step 1

```
> kubectl get ns
NAME                      STATUS        AGE
cattle-system             Terminating   25m
cattle-ui-plugin-system   Active        24m
default                   Active        37m
gke-managed-cim           Active        36m
gke-managed-system        Active        36m
gmp-public                Active        36m
gmp-system                Active        36m
kube-node-lease           Active        37m
kube-public               Active        37m
kube-system               Active        37m
local                     Active        24m

> kubectl get all -n cattle-system
NAME                                   READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/cattle-cluster-agent   0/1     0            0           25m

> kubectl describe ns cattle-system
Name:         cattle-system
Labels:       kubernetes.io/metadata.name=cattle-system
Annotations:  management.cattle.io/no-default-sa-token: true
Status:       Terminating
Conditions:
  Type                                         Status  LastTransitionTime               Reason                Message
  ----                                         ------  ------------------               ------                -------
  NamespaceDeletionDiscoveryFailure            False   Mon, 04 Nov 2024 16:28:44 +0530  ResourcesDiscovered   All resources successfully discovered
  NamespaceDeletionGroupVersionParsingFailure  False   Mon, 04 Nov 2024 16:28:44 +0530  ParsedGroupVersions   All legacy kube types successfully parsed
  NamespaceDeletionContentFailure              False   Mon, 04 Nov 2024 16:28:45 +0530  ContentDeleted        All content successfully deleted, may be waiting on finalization
  NamespaceContentRemaining                    True    Mon, 04 Nov 2024 16:28:44 +0530  SomeResourcesRemain   Some resources are remaining: deployments.apps has 1 resource instances
  NamespaceFinalizersRemaining                 True    Mon, 04 Nov 2024 16:28:44 +0530  SomeFinalizersRemain  Some content in the namespace has finalizers remaining: wrangler.cattle.io/legacy-k3sBasedUpgrader-deprecation in 1 resource instances
```

**Comment by snasovich** [comment]
Per investigations completed by @cpinjani @yiannistri the namespace deletion is blocked by `wrangler.cattle.io/legacy-k3sBasedUpgrader-deprecation` finalizer which I guess was added as part of https://github.com/rancher/rancher/issues/42448.
@jiaqiluo , please take a look into it with priority.

**Comment by jiaqiluo** [comment]
The issue is reproduced on a Rancher `2.10.0-alpha6` Docker installation setup. 

## Causes

In the downstream cluster, the `systemChart` handler runs within the cattle-cluster-agent, adding the `wrangler.cattle.io/legacy-k3sBasedUpgrader-deprecation` finalizer to all Deployments in all namespaces. When an imported cluster is removed from Rancher, Rancher triggers a cleanup job on the downstream cluster to delete all Rancher-related resources. However, due to this finalizer, Deployments in the `cattle-system` namespace, as well as the namespace itself, become stuck in the removal process. This occurs even though the cleanup job reports success because it successfully initiates the namespace deletion.


## Fixes

The cleanup job should remove the finalizer from all Deployments across all namespaces before proceeding with the deletion of those namespaces.

**Comment by jiaqiluo** [comment]
this issue can be validated on the latest head image

**Comment by valaparthvi** [comment]
Fixed on v2.10-66ce2f133c38379570f59276799bebc222d9ef2d-head.
It works as expected for all the 3 hosted providers EKS, GKE, and AKS.

**Comment by valaparthvi** [comment]
Shouldn't all Rancher related namespaces be deleted? I still see `cattle-ui-plugin-system` after the imported cluster is deleted.

**Comment by cpinjani** [comment]
> Shouldn't all Rancher related namespaces be deleted? I still see `cattle-ui-plugin-system` after the imported cluster is deleted.

There is separate issue for it https://github.com/rancher/rancher/issues/47883
